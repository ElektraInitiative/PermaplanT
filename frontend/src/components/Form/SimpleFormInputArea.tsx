import { HTMLInputTypeAttribute } from 'react';
import { FieldValues, Path, UseFormRegister } from 'react-hook-form';

interface SimpleFormInputAreaProps<T extends FieldValues> extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {
  id: Path<T>; 
  labelText: string;
  register?: UseFormRegister<T>;
}

export default function SimpleFormAreaInput<T extends FieldValues> ({
  id, 
  labelText = '',
  register,
 ...props
}: SimpleFormInputAreaProps<T>) {
  // Extract the input fields value from the respective Events before calling onChange.
/*  const callOnChange = function <E>(event: React.ChangeEvent<E> | React.KeyboardEvent<E>) {
    if (onChange == null) return;

    // If somebody finds a way of fixing the next line, please do!
    const value = (event.target as unknown as HTMLInputElement).value;

    if (type === 'number') {
      onChange(parseInt(value));
    }

    onChange(value);
  }; */

  return (
    <div className="dark:text-white">
      <label htmlFor={id} className="mb-2 block text-sm font-medium">
        {labelText}
        {props.required ? <span className="text-red-800"> *</span> : <></>}
      </label>
      <textarea
        id={id}
        {...props}
        {...register?.(id)}
        className="dark:bg-primary-textfield-dark block w-full rounded-lg border border-neutral-500 bg-neutral-100 p-2.5 text-sm placeholder-neutral-500 focus:border-primary-500 focus:outline-none dark:bg-neutral-100-dark dark:focus:border-primary-300"
      />
    </div>
  );
}
