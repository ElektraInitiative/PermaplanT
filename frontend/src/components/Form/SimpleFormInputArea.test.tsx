import SimpleFormInputArea from './SimpleFormInputArea';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <SimpleFormInputArea id="test" labelText="Some Label" />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
