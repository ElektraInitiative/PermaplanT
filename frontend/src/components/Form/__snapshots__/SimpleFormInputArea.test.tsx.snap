// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="dark:text-white"
>
  <label
    className="mb-2 block text-sm font-medium"
    htmlFor="test"
  >
    Some Label
  </label>
  <textarea
    className="dark:bg-primary-textfield-dark block w-full rounded-lg border border-neutral-500 bg-neutral-100 p-2.5 text-sm placeholder-neutral-500 focus:border-primary-500 focus:outline-none dark:bg-neutral-100-dark dark:focus:border-primary-300"
    id="test"
  />
</div>
`;
